build.rs

```diff
@@ -148,12 +165,21 @@ fn main() {
     println!("cargo:rerun-if-env-changed=LIBPCAP_LIBDIR");
     println!("cargo:rerun-if-env-changed=LIBPCAP_VER");
 
-    let mut libdirpath: Option<PathBuf> = None;
-    if let Ok(libdir) = env::var("LIBPCAP_LIBDIR") {
+    // If user explicitly set LIBPCAP_LIBDIR, honour their wishes. This keeps
+    // existing build scripts running. If it's not set, try pkg-config. If
+    // that's not set, try last ditch effort to build even though library wasn't
+    // explicitly given.
+    let version = if let Ok(libdir) = env::var("LIBPCAP_LIBDIR") {
         println!("cargo:rustc-link-search=native={}", libdir);
-        libdirpath = Some(PathBuf::from(&libdir));
-    }
+        get_libpcap_version(Some(PathBuf::from(&libdir))).unwrap()
+    } else if let Ok(library) = pkg_config::Config::new()
+        .atleast_version(&Version::LOWEST_SUPPORTED.to_string())
```

Comment:
I wonder if we should check for LIBPCAP_VER here and only default to LOWEST_SUPPORTED if it's unset. This would allow "picking" a version

Reply:
I decided to do this. We can revert it if it's dumb.